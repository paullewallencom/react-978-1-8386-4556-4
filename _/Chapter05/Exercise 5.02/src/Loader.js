import React from "react";

// styles
import "./loader.css";

const Loader = props => {
  const { spinnerSize, isVisible } = props;
  if (!isVisible) {
    return null;
  }
  return (
    <div className="spinner">
      <svg
        focusable="false"
        width={spinnerSize}
        height={spinnerSize}
        viewBox={`0 0 ${spinnerSize} ${spinnerSize}`}
      >
        <circle
          cx={spinnerSize / 2}
          cy={spinnerSize / 2}
          r={spinnerSize / 2 - 10}
        />
      </svg>
    </div>
  );
};

export default Loader;
