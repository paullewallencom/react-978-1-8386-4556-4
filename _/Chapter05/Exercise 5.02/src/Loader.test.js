import React from "react";
import { render, cleanup } from "@testing-library/react";
import Loader from "./Loader";

afterEach(cleanup);

it("renders loader", () => {
  const { asFragment } = render(<Loader spinnerSize={50} isVisible={true} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders null when isVisible is set to false", () => {
  const { container } = render(<Loader spinnerSize={50} isVisible={false} />);
  expect(container.firstChild).toBeNull();
});

it("sets the right svg size", () => {
  const spinnerSize = 100;
  const { container } = render(
    <Loader spinnerSize={spinnerSize} isVisible={true} />
  );
  expect(container.querySelector("svg").getAttribute("height")).toBe(
    spinnerSize.toString()
  );
  expect(container.querySelector("svg").getAttribute("width")).toBe(
    spinnerSize.toString()
  );
});
