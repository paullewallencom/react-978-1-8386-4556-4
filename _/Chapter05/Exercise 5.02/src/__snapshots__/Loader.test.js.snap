// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  >
    <svg
      focusable="false"
      height="50"
      viewBox="0 0 50 50"
      width="50"
    >
      <circle
        cx="25"
        cy="25"
        r="15"
      />
    </svg>
  </div>
</DocumentFragment>
`;
